### ISSUE TITLE
ArchLinux - steam window not showing using nvidia gpu

### ISSUE THREAD

**Issue by KS-HTK** [ValveSoftware/steam-for-linux#8509]
#### Your system information

* Steam client version (build number or date): Can't access that Info due to the Issue (steam.sh[2556]: Steam runtime environment up-to-date!)
* Distribution (e.g. Ubuntu): ArchLinux (Zen & default Kernel) DM: Gnome
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Additional system information

* nvidia-dkms version: 510.54-1
* nvidia-utils version: 510.54-1
* lib32-nvidia-utils version: 510.54-1

#### Issue Description:
I recently configured my Zen Kernel to use the dedicated NVidia GPU of my Laptop, same as I have it configured for the default Kernel. Thus moved from wayland back to X11.
But although working on the default Kernel just a few weeks ago and working on the Intel Integrated Graphics only Hours ago steam will launch but not open a window.
Starting it trough terminal showed no errors at first.
After running `steam-runtime --reset` I can get the initial update window, but wont get the normal Steam window to show up.
After a bit of googling for a solution people mentioned to make sure `lib32-nvidia-utils` version should match the `nvidia` (`nvidia-dkms` in case of Zen Kernel) version, but this is the case on my System. So I am not sure what the problem is exactly, guessing it is related to the GPU Driver.
Interestingly I could run steam on the NVidia card prior to changing the whole system to only use the NVidia GPU by using `prime-run steam`. (Tested that to make sure it would work before re configuring Gnome to use X11)

#### Steps for reproducing this issue:

1. Update ArchLinux on System using NVidia GPU
2. Install steam
3. try start steam
4. observe

#### Terminal output:
(after running `steam-runtime --reset`)
```
steam.sh[2949]: Running Steam on arch rolling 64-bit
steam.sh[2949]: STEAM_RUNTIME is enabled automatically
setup.sh[3021]: Steam runtime environment up-to-date!
steam.sh[2949]: Can't find 'steam-runtime-check-requirements', continuing anyway
[2022-03-29 00:03:01] Startup - updater built Nov 22 2021 22:06:59
[2022-03-29 00:03:01] Opted in to client beta 'publicbeta' via beta file
You are in the 'publicbeta' client beta.
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
Installing breakpad exception handler for appid(steam)/version(1647446817)
SteamUpdateUI: An X Error occurred
X Error of failed request:  BadValue (integer parameter out of range for operation)
Major opcode of failed request:  156 (NV-GLX)
Minor opcode of failed request:  6
Value in failed request:  0x0
Serial number of failed request:  35
xerror_handler: X failed, continuing
SteamUpdateUI: An X Error occurred
X Error of failed request:  BadValue (integer parameter out of range for operation)
```

**Comment by kisak-valve** [comment]
Hello @KS-HTK, can you check if any 32 bit OpenGL application like `glxinfo32` from `lib32-mesa-utils` works?

**Comment by KS-HTK** [comment]
Ok. `lib32-mesa-utils` was not installed. So I installed that and tried to run `glxinfo32`.
This gives a similar error. So I guess it is related to running 32-bit applications then?

```
name of display: :1
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  156 (NV-GLX)
  Minor opcode of failed request:  6 ()
  Value in failed request:  0x0
  Serial number of failed request:  96
  Current serial number in output stream:  96
```

**Comment by kisak-valve** [comment]
Yes, that eliminates Steam from the picture and tells you that there's a general issue with OpenGL happening on your system. I would double check that the NVIDIA driver's running kernel module, 32 bit userspace libraries, and 64 bit userspace libraries all have exactly the same version.

**Comment by KS-HTK** [comment]
Ok, thanks. I just found out some other interesting bit of info. Running `sudo glxinfo32` does not error and gives a lot of output, so I guess it is nothing to do with the libraries itself but with permission management. The problem is that as far as I can tell the group `video` needs to be given to the user, which it already was. And I had it working before…

But ok, I will try and debug it further. Thanks for the help.

#### Output of `sudo glxinfo32`
```
name of display: :1
display: :1  screen: 0
direct rendering: Yes
server glx vendor string: SGI
server glx version string: 1.4
server glx extensions:
[…]
```
